v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -0,0 +1,647 @@
+        const command = "npx hardhat task subtask --param <paramValue>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, SUBTASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, {
+          param: "<paramValue>",
+        });
+      });
+
+      it("should get the tasks and its parameter as type boolean with value set to true", function () {
+        const command = "npx hardhat task --flag";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, TASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true]);
+        assert.deepStrictEqual(res.taskArguments, { flag: true });
+      });
+
+      it("should throw because the parameter when set as bool (flag behavior) should not be followed by a bool value", function () {
+        const command = "npx hardhat task --flag false";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, TASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, { flag: false });
+      });
+
+      it("should convert on the fly the camelCase parameter to kebab-case", function () {
+        // Parameter with name "optionFlag" should be converted on the fly to "--option-flag"
+        const command = "npx hardhat task --option-flag <value>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        const res = parseTaskAndArguments(
+          cliArguments,
+          usedCliArguments,
+          hre,
+        ) as TaskRes;
+
+        assert.equal(res.task.id, TASK.id);
+        assert.deepStrictEqual(usedCliArguments, [true, true, true]);
+        assert.deepStrictEqual(res.taskArguments, { optionFlag: "<value>" });
+      });
+
+      it("should throw because the parameter is not defined", async function () {
+        const command = "npx hardhat task --undefinedParam <value>";
+
+        const cliArguments = command.split(" ").slice(2);
+        const usedCliArguments = [false, false, false];
+
+        assert.throws(
+          () => parseTaskAndArguments(cliArguments, usedCliArguments, hre),
+          new HardhatError(
```

Comment:
I think an easier syntax may be using an object with the error and the message: 
```
{
  name: "HardhatError",
  message: 'Invalid parameter "--undefinedParam". It is neither a valid global parameter nor associated with any task. Did you forget to add the task first, or did you misspell it?',
}
```

Reply:
This will show the message in the test, true, but then all the test should be modified if the error message is changed. 
Wdyt?